### ISSUE TITLE
AIP-44 Migrate TaskInstance._run_raw_task  to Internal API

### ISSUE THREAD

**Issue by mhenc** [apache/airflow#29312]
Prerequisite:
https://github.com/apache/airflow/issues/29311

Link:
https://github.com/apache/airflow/blob/main/airflow/models/taskinstance.py#L1351

This metho adds/updates multiple objects (TaskInstance, Log).
It also calls some other methods passing session.  
We may simply migrate all method called there or (preferred) refactor the method to have multiple phases :
Internal API call - Local execution - internal API call .... 

It is preferred as the whole sections there are executed in a transaction.

**Comment by bugraoz93** [comment]
I would like to work on this after #29311. Of course, if there is no one who is already a volunteer.

**Comment by mhenc** [comment]
Great. Currently it is blocked on @potiuk PoC in which we don't need to serialize TaskInstance. 
https://github.com/apache/airflow/pull/29513#discussion_r1112952981

**Comment by potiuk** [comment]
Correct. Still in-progress. Making it small enough change to not impact "regular" cases and also clear enough to warrant db-less mode requires quite some thiking on how to do it best :)

**Comment by bugraoz93** [comment]
Thanks for pointing that out mhenc!

No pressure potiuk, looking forward to seeing it :)
